src/response.rs

```diff
@@ -68,9 +69,23 @@ impl Response {
             ext: Extensions::new(),
             peer_addr: None,
             local_addr: None,
+            host: None,
         }
     }
 
+    /// Sets the host that this response is from, using
+    /// [Request::host](crate::Request::host)
+    pub fn set_host_from_request(&mut self, request: &crate::Request) {
+        self.host = request.host().map(|h| h.to_string())
+    }
```

Comment:
I think having a `Response::set_host` method might be a better fit; it should be about the same length to copy it from a request, but much easier:

```rust
let mut req = ...;
let mut res = ...;

res.set_host(req.host());